addons/mass_mailing/models/mass_mailing.py

```diff
@@ -133,6 +133,23 @@ def _compute_contact_nbr(self):
         for mailing_list in self:
             mailing_list.contact_nbr = data.get(mailing_list.id, 0)
 
+    @api.multi
+    def write(self, vals):
+        self.ensure_one()
+
+        # Prevent archiving used mailing list
+        if 'active' in vals and not vals.get('active'):
+            mass_mailings = self.env['mail.mass_mailing'].search_read([
+                ('state', '!=', 'done')
```

Comment:
Shouldn't this work instead of filtering afterwards?
```suggestion
                ('state', '!=', 'done'), ('contact_list_ids', 'in', self.ids)
```

Reply:
Indeed, but I still need to recompute a list for outputting the used mailing names